db/database.go

```diff
@@ -569,7 +569,13 @@ func (context *DatabaseContext) RemoveObsoleteDesignDocs(previewOnly bool) (remo
 
 // Removes previous versions of Sync Gateway's design docs found on the server
 func (context *DatabaseContext) RemoveObsoleteIndexes(previewOnly bool) (removedIndexes []string, err error) {
-	return removeObsoleteIndexes(context.Bucket, previewOnly, context.UseXattrs())
+
+	gocbBucket, ok := base.AsGoCBBucket(context.Bucket)
+	if !ok {
+		return nil, errors.New("Must be gocbBucket")
```

Comment:
This is changing the behaviour of the function.
Previously we just warned and skipped over when it was a non-gocb bucket (from `removeObsoleteIndexes`)

```go
	gocbBucket, ok := base.AsGoCBBucket(bucket)
	if !ok {
		base.Warnf(base.KeyAll, "Cannot remove obsolete indexes for non-gocb bucket - skipping.")
		return removedIndexes, nil
	} 
```